src/coreComponents/fieldSpecification/FieldSpecificationManager.cpp

```diff
@@ -165,21 +170,72 @@ void FieldSpecificationManager::validateBoundaryConditions( MeshLevel & mesh ) c
     }
 
     // Step 4: issue an error or a warning if the field was not found
-
     // if all sets are missing, we stop the simulation.
     if( areAllSetsMissing )
     {
       // loop again over the map to collect the set names
       string_array missingSetNames;
+
       for( auto const & mapEntry : isTargetSetCreated )
       {
         missingSetNames.emplace_back( mapEntry.first );
       }
-      GEOS_THROW( GEOS_FMT( "\n{}: there is/are no set(s) named `{}` under the {} `{}`.\n",
-                            fs.getWrapperDataContext( FieldSpecificationBase::viewKeyStruct::objectPathString() ),
-                            fmt::join( missingSetNames, ", " ),
-                            FieldSpecificationBase::viewKeyStruct::objectPathString(), fs.getObjectPath() ),
-                  InputError );
+
+      string setNamesError = GEOS_FMT( "\n{}: there is/are no set(s) named `{}` under the {} `{}`.\n",
+                                       fs.getWrapperDataContext( FieldSpecificationBase::viewKeyStruct::objectPathString() ),
+                                       fmt::join( missingSetNames, ", " ),
+                                       FieldSpecificationBase::viewKeyStruct::objectPathString(), fs.getObjectPath() );
+
+      Group const & problemManager = this->getGroupByPath( "/Problem" );
+      string const & objPath = fs.getObjectPath();
+
+      string setNamesAvailable;
+
+      if( objPath == "nodeManager" )
+      {
+        setNamesAvailable.append( "Set names available are: xneg ,yneg ,zneg ,xpos ,ypos, " );
```

Comment:
```suggestion
        setNamesAvailable.append( "Set names available are: xneg, yneg, zneg, xpos, ypos, " );
```
and same for `",rneg ,rpos ,tneg ,tpos"`

Reply:
those hardcoded names need to be defined somewhere in one place...

Reply:
Can you add an instance of a generated error for an InternalMesh case so we can confirm the `xpos` & co are visible? Then I will resolve this conversation.